app/Http/Controllers/PaymentController.php

```diff
@@ -35,10 +35,19 @@ public function index()
      */
     public function store(Request $request)
     {
+        $currentUser = auth()->user();
+
+        if (is_null($request->recorded_by) ||
```

Comment:
Consider using `!$request->has('recorded_by')` here. As far as I can tell it accomplishes the same thing you're doing here, it's just a more Laravel-esque way of checking.